.github/workflows/docs-update.yml

```diff
@@ -0,0 +1,70 @@
+name: Update Docs
+
+on:
+  push:
+    tags:
+      # https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string -- github doesn't like this regex??
+      - v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)
+      - v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-|\+)([a-zA-Z0-9]|-|+|\.)*
+
+permissions:
+  contents: write
+  id-token: write
+
+concurrency:
+  group: Update Docs
+  cancel-in-progress: false
+
+jobs:
+  update_docs:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout the docs
+        uses: actions/checkout@v3
+        with:
+          ref: gh-pages
+
+      - name: Save old docs
+        run: tar -cvf /tmp/old-docs.tar ./docs/generated
+
+      - name: Checkout tag
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.ref_name }}
+          clean: false #  doesn't seem to work?? Hense use of tar.
```

Comment:
[Question] What happened when you tried?

Reply:
The same thing as when I don't